pkg/avancement/service_manager.go

```diff
@@ -102,7 +104,17 @@ func (s *ServiceManager) Promote(serviceName, fromURL, toURL, newBranchName stri
 func (s *ServiceManager) checkoutSourceRepo(repoURL, branch string) (git.Repo, error) {
 	repo, err := s.cloneRepo(repoURL, branch)
 	if err != nil {
-		return nil, fmt.Errorf("failed to clone source repo %s: %w", repoURL, err)
+		/*
+			We don't want to surface any tokens here, without this code one can get:
+			failed to clone repo https://promotion:<TOKEN>@github.com/org/repo: exit status 128
+			So if the status is a 128, provide some useful debug steps.
+		*/
+		errorToDisplay := err
+		if strings.Contains(err.Error(), "exit status 128") {
+			errorToDisplay = errors.New(
+				"check that the access token used is correct with repository level permissions and that a target branch with the desired name does not already exist on the staging repository")
```

Comment:
You could even add the URL of the staging repo to this error message, as well as maybe splitting it over a few lines : )

Reply:
Nice idea, thanks will look into it - yeah, sharing the same concern about having to take out the token for that bit too actually (e.g. in the case of a private repository)